test/schema/simple.yaml

```diff
@@ -0,0 +1,34 @@
+---
+apiVersion: beehive/v1
+kind: Workflow
+metadata:
+  name: beehive-deployment
+  labels:
+    app: example2
+spec:
+  strategy:
+    type: sequence
+  template:
+    metadata:
+      name: beehive-deployment
+      labels:
+        app: example
+        use-case: test
+  agents:
+  - agent1
+  - agent2
+  - agent3
+  exception: step4
+  steps:
+  - step1
+  - step2
+  - step3
+  - step4
+  step1:
+    agent: agent1
+  step2:
+    agent: agent2
+  step3:
+    agent: agent1
+  step4:
+    agent: agent3
```

Comment:
Instead of hardcoding the step names, could we do something like this instead?

```yaml
steps:
- name: step1
  agent: agent1
- name: step2
  agent: agent2
```

Reply:
Agree. It also makes it easier to then add additional attributes for each step.

It also collapses the steps array (which defines the sequence) and the distinct step definitions in the original proposal, which feels as if it involves duplication

Reply:
OK. But my guess we will need both or a way to have steps separated at some point. But for now, since use cases are relatively simple, we can do this.